drivers/display/display_ili9342c.h

```diff
@@ -0,0 +1,121 @@
+	uint8_t gamset[ILI9342C_GAMSET_LEN];
+	uint8_t ifmode[ILI9342C_IFMODE_LEN];
+	uint8_t frmctr1[ILI9342C_FRMCTR1_LEN];
+	uint8_t invtr[ILI9342C_INVTR_LEN];
+	uint8_t disctrl[ILI9342C_DISCTRL_LEN];
+	uint8_t etmod[ILI9342C_ETMOD_LEN];
+	uint8_t pwctrl1[ILI9342C_PWCTRL1_LEN];
+	uint8_t pwctrl2[ILI9342C_PWCTRL2_LEN];
+	uint8_t pwctrl3[ILI9342C_PWCTRL3_LEN];
+	uint8_t vmctrl1[ILI9342C_VMCTRL1_LEN];
+	uint8_t setextc[ILI9342C_SETEXTC_LEN];
+	uint8_t pgamctrl[ILI9342C_PGAMCTRL_LEN];
+	uint8_t ngamctrl[ILI9342C_NGAMCTRL_LEN];
+	uint8_t ifctl[ILI9342C_IFCTL_LEN];
+};
+
+/* Initializer macro for ILI9342C registers. */
+#define ILI9342c_REGS_INIT(n)									\
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), gamset) == ILI9342C_GAMSET_LEN, \
+		     "ILI9342c: Error length of gamma set (GAMSET) register");                    \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), ifmode) == ILI9342C_IFMODE_LEN, \
+		     "ILI9342c: Error length of frame rate control (IFMODE) register");           \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), frmctr1) == ILI9342C_FRMCTR1_LEN, \
+		     "ILI9342c: Error length of frame rate control (FRMCTR1) register");          \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), invtr) == ILI9342C_INVTR_LEN,   \
+		     "ILI9342c: Error length of Display Inversion Control (INVTR) register");     \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), disctrl) == ILI9342C_DISCTRL_LEN,    \
+		     "ILI9342c: Error length of display function control (DISCTRL) register");    \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), etmod) == ILI9342C_ETMOD_LEN,   \
+		     "ILI9342c: Error length of Entry Mode set (ETMOD) register");         \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), pwctrl1) == ILI9342C_PWCTRL1_LEN,    \
+		     "ILI9342c: Error length of power control 1 (PWCTRL1) register");             \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), pwctrl2) == ILI9342C_PWCTRL2_LEN,    \
+		     "ILI9342c: Error length of power control 2 (PWCTRL2) register");             \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), pwctrl3) == ILI9342C_PWCTRL3_LEN,    \
+		     "ILI9342c: Error length of power control 3 (PWCTRL3) register");             \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), vmctrl1) == ILI9342C_VMCTRL1_LEN,    \
+		     "ILI9342c: Error length of VCOM control 1 (VMCTRL1) register");              \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), pgamctrl) == ILI9342C_PGAMCTRL_LEN,  \
+		     "ILI9342c: Error length of positive gamma correction (PGAMCTRL) register");  \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), ngamctrl) == ILI9342C_NGAMCTRL_LEN,  \
+		     "ILI9342c: Error length of negative gamma correction (NGAMCTRL) register");  \
+	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9342c), ifctl) == ILI9342C_IFCTL_LEN,   \
+		     "ILI9342c: Error length of frame rate control (IFCTL) register");            \
```

Comment:
None of this makes any sense.

Reply:
it is like what is already here: https://github.com/zephyrproject-rtos/zephyr/blame/main/drivers/display/display_ili9341.h

Reply:
Still does no convince me to litter the source code with BUILD_ASSERT and does not help you.

Reply:
Done